cloudwatch_logs_common/src/utils/log_file_manager.cpp

```diff
@@ -77,16 +70,30 @@ FileObject<LogCollection> LogFileManager::readBatch(
       log_data.push_front(input_event);
       data_tokens.push_back(new_data_token);
     }
+    else if(file_manager_strategy_->discardOldLogs() && latestTime - curTime > TWO_WEEK_IN_SEC){
```

Comment:
That is true but shouldn't we be going through the rest of the logs to discard those older than 2 weeks?

Reply:
If we could delete the data after submitting the task then we are prioritizing sending data back over cleanup (since the inputs have been sanitized). I'm wondering if this is possible as a post task submission job.

Reply:
This makes sense but I think if this function has already gone through the process of sorting the logs by timestamp then it might be valuable to retain that effort and quickly process the rest of it rather than leaving the unsorted logs in the file and slowing down subsequent readBatch commands.

Reply:
You can always think outside of the scope of the function: you already have the tokens needed to delete a file. One way to accomplish this while being mindful of the `FileUploadStreamer::work` method is this:

- `readBatch` obtains file tokens to be deleted, stores them locally, and returns the relevant `FileObject` to be uploaded. 
- FileUploadStreamer::work` converts the `FileObject` to a `FileUploadTask` and submits into the queue to be uploaded
- after submitting the `FileUploadTask` the FileUploadStreamer requests the `data_reader_` to clear stale data, which makes sense to do here because the next action after submitting the task is waiting for the upload to complete. This way the file I/O does not block the current upload task, but only future tasks.